2to3.patch

```diff
@@ -1,270 +1,57 @@
-+        print("  Restoring: %s" % self.filepath)
-         input_filename = self.filepath
-         inputfile = self.storage.read_file(input_filename)
-         if self.get_extension(input_filename) == '.gpg':
-@@ -77,7 +77,7 @@
-             uncompressed_file = self.uncompress_file(inputfile)
+@@ -94,7 +94,7 @@
              inputfile.close()
              inputfile = uncompressed_file
--        print "  Restore tempfile created: %s" % utils.handle_size(inputfile)
-+        print("  Restore tempfile created: %s" % utils.handle_size(inputfile))
-         self.dbcommands.run_restore_commands(inputfile)
- 
-     def get_extension(self, filename):
-@@ -99,8 +99,8 @@
-         """ Unencrypt this file using gpg. The input and the output are filelike objects. """
+         print("  Restore tempfile created: %s" % utils.handle_size(inputfile))
+-        cont = input("Are you sure you want to continue? [Y/n]")
++        cont = eval(input("Are you sure you want to continue? [Y/n]"))
+         if cont.lower() not in ('y', 'yes', ''):
+             print("Quitting")
+             sys.exit(0)
+@@ -123,7 +123,7 @@
          import gnupg
          def get_passphrase():
--            print 'Input Passphrase: '
--            return raw_input()
-+            print('Input Passphrase: ')
-+            return input()
+             print('Input Passphrase: ')
+-            return input()
++            return eval(input())
          
-         temp_dir = tempfile.mkdtemp()
-         try:
-@@ -129,7 +129,7 @@
- 
-     def list_backups(self):
-         """ List backups in the backup directory. """
--        print "Listing backups on %s in /%s:" % (self.storage.name, self.storage.backup_dir())
-+        print("Listing backups on %s in /%s:" % (self.storage.name, self.storage.backup_dir()))
-         for filepath in self.storage.list_directory():
--            print "  %s" % os.path.basename(filepath)
--            print utils.filename_details(filepath)
-+            print("  %s" % os.path.basename(filepath))
-+            print(utils.filename_details(filepath))
---- dbbackup/storage/dropbox_storage.py	(original)
-+++ dbbackup/storage/dropbox_storage.py	(refactored)
-@@ -4,7 +4,7 @@
- import pickle
- import os
- import tempfile
--import cStringIO
-+import io
- from django.conf import settings
- from dropbox import session
- from dropbox.client import DropboxClient
-@@ -105,7 +105,7 @@
-         ignore_404 = kwargs.pop("ignore_404", False)
+         temp_dir = tempfile.mkdtemp(dir=dbbackup_settings.TMP_DIR)
          try:
-             response = method(*args, **kwargs)
--        except ErrorResponse, e:
-+        except ErrorResponse as e:
-             if ignore_404 and e.status == 404:
-                 return None
-             errmsg = "ERROR %s" % (e,)
-@@ -189,7 +189,7 @@
-     def chunked_file(filehandle, chunk_size=FILE_SIZE_LIMIT):
-         eof = False
-         while not eof:
--            tmpfile = cStringIO.StringIO()
-+            tmpfile = io.StringIO()
```

Comment:
Just realized that these lines was removed from patch.
Well, the better way will be test this new code with python 3 which I don't have right now.